opennms-dao/src/main/java/org/opennms/netmgt/dao/hibernate/InterfaceToNodeCacheDaoImpl.java

```diff
@@ -206,21 +216,28 @@ public InterfaceToNodeCacheDaoImpl(long refreshRate) {
 
     @PostConstruct
     public void init() {
-        // we call this, to ensure the bean-initialization is blocked, until it is initialized
-        dataSourceSync();
-
-        if (refreshRate > 0) {
-            refreshTimer.schedule(new TimerTask() {
-                @Override
-                public void run() {
-                    try {
-                        dataSourceSync();
-                    } catch (Exception ex) {
-                        LOG.error("An error occurred while synchronizing the datasource: {}", ex.getMessage(), ex);
+        // sync datasource asynchronously in order to not block bean initialization.
+        syncDataSourceAsynchronously().whenComplete((result, ex) -> {
+            initialNodeSyncDone.countDown();
+            if (refreshRate > 0) {
+                refreshTimer.schedule(new TimerTask() {
+                    @Override
+                    public void run() {
+                        try {
+                            dataSourceSync();
```

Comment:
I have a feeling this is part of the actual expected behavior (I'm not familiar with it) just wondering if is meant to call the dataSourceSync again when refreshRate > 0?

Reply:
yes, this will refresh every 5 mins by this [system property](https://github.com/OpenNMS/opennms/blob/5838d88eb0a0e8ad1a9011a3a142ccc9bc37e1eb/opennms-dao/src/main/resources/META-INF/opennms/applicationContext-shared.xml#L141).